### PR TITLE
Iss hipm 796 - Achievement notifications

### PR DESCRIPTION
**Pull request by Resudo**
This branch adds the possibility to display notifications for achievements. To display achievements the function `QueueAchievementNotifications(..)` has to be called from within a `NavigationViewModel`.

If you don't want to test this functionality with real achievements you can use the following code sniplet by adding it to any function you can intentionally trigger in a `NavigationViewModel` (e.g. `NavigateToExhibiDetails(..)` in the `ExhibitsOverviewViewModel`):

```
var collection = new ObservableCollection<IAchievement> {
    new ExhibitsVisitedAchievement {
        Title = "The exhibit visitor",
        Description = "Visit an exhibit for the first time",
        ThumbnailUrl = "achievements/api/image/0/"}};
AchievementNotification.QueueAchievementNotifications(collection);
```

Minimizing the app pauses the notifications and landscape layout is supported as well.

_Dismissing a notification on iOS does not work because the input transparency behaves differently._

### CODE REVIEW

File: Sources/HipMobileUI/Pages/AppetizerPage.xaml [hunk 1]
```diff
@@ -18,51 +18,56 @@
...
+                                    <Label Text="1.222 Bewertungen" FontSize="10" HorizontalOptions="Center" VerticalOptions="Center" />
+                                    <Label Text="★★★★☆ " FontSize="10" HorizontalOptions="Center" VerticalOptions="Center" />
+                                </StackLayout>
+                            </Grid>
+                        </ScrollView>
 
-            </Grid>
-            <Button AbsoluteLayout.LayoutBounds=".5,1,.95,.1" AbsoluteLayout.LayoutFlags="All"
+                        <Label Text="1/10" FontAttributes="Bold" FontSize="20" HorizontalOptions="End" />
+
```
Comment (nataliefriese) [inline Sources/HipMobileUI/Pages/AppetizerPage.xaml]: is the line without content (62) required

File: Sources/HipMobileUI/Pages/AppetizerPage.xaml [hunk 2]
```diff
@@ -18,51 +18,56 @@
...
+                    </Grid>
```
Comment (nataliefriese) [inline Sources/HipMobileUI/Pages/AppetizerPage.xaml]: is this one well formated?

File: Sources/HipMobileUI/Pages/CharacterSelectionPage.xaml [hunk 3]
```diff
@@ -1,37 +1,58 @@
...
-                <Grid.GestureRecognizers>
-                    <TapGestureRecognizer Command="{Binding ProfessorGridTappedCommand}" />
-                </Grid.GestureRecognizers>
-                <Grid.RowDefinitions>
-                    <RowDefinition Height="0.7*" />
-                    <RowDefinition Height="0.3*" />
-                </Grid.RowDefinitions>
-                <Image x:Name="ProfessorImage" Grid.Row="0" Source="ic_professor" Aspect="AspectFit" Margin="10" />
-                <Label x:Name="ProfessorText" Grid.Row="1"
+                    </Grid>
```
Comment (nataliefriese) [inline Sources/HipMobileUI/Pages/CharacterSelectionPage.xaml]: format

File: Sources/HipMobileUI/Pages/AppetizerPage.xaml (hunk 2 above)
Comment (Resudo) [reply Sources/HipMobileUI/Pages/AppetizerPage.xaml]: See comment above.

File: Sources/HipMobileUI/Pages/CharacterSelectionPage.xaml (hunk 3 above)
Comment (Resudo) [reply Sources/HipMobileUI/Pages/CharacterSelectionPage.xaml]: Before the PR I went through all the files modified by me and applied our formatting so it should be all right. But I will check again and also remove the blank lines.

File: Sources/HipMobileUI/ViewModels/NavigationViewModel.cs [hunk 4]
```diff
@@ -44,10 +47,11 @@ public virtual void OnHidden()
         }
```
Comment (nataliefriese) [inline Sources/HipMobileUI/ViewModels/NavigationViewModel.cs]: this method does nothing?

**Comment by nataliefriese** [comment]
code looks good to me 
tomorrow Smriti and me will check if it works fine

File: Sources/HipMobileUI/ViewModels/NavigationViewModel.cs (hunk 4 above)
Comment (Resudo) [reply Sources/HipMobileUI/ViewModels/NavigationViewModel.cs]: In this class, yes. But some of the classes inheriting from it override this method.

**Pull request merged**